src/editor/event.cpp

```diff
@@ -0,0 +1,73 @@
+#include "event.h"
+
+QString Event::sTabs="";
+bool Event::sIsSleshPressed=false;
+void Event::eventInsertSymbol(CodeEditor *codeEditor, QKeyEvent *e, QString s)
+{
+        plainTextPressEvent(codeEditor, e);
+        codeEditor->insertPlainText(s);
+        for (int i = 0; i < s.size(); i++)
+        {
+            codeEditor->moveCursor(QTextCursor::Left);
+        }
+}
+
+bool Event::isinsidebracket(CodeEditor *codeEditor)
+{
+    QTextCursor cursor = codeEditor->textCursor();
+    if(cursor.position() != codeEditor->document()->toPlainText().size())
+    {
+        cursor.movePosition(QTextCursor::PreviousCharacter);
+        int currentpos = cursor.position();
+        QString prev = codeEditor->document()->toPlainText().at(currentpos);
+        cursor.movePosition(QTextCursor::NextCharacter);
+        currentpos = cursor.position();
+        QString next = codeEditor->document()->toPlainText().at(currentpos);
+        if(prev == "{" && next == "}")
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+void Event::plainTextPressEvent(CodeEditor *codeEditor, QKeyEvent *e)
+{
+    codeEditor->QPlainTextEdit::keyPressEvent(e);
+}
+
+void Event::autotab(CodeEditor *code)
+{
```

Comment:
autotab is a slot of class Event
However in method void EventEnter::operator()(CodeEditor *codeEditor, QKeyEvent *e) you emit autotab
Only signal can be emitted
Howerer you can simply call slot as regular class method without emit
So if you call: autotab(codeEditor); it will work